File: 백준/Silver/2422. 한윤정이 이탈리아에 가서 아이스크림을 사먹는데/한윤정이 이탈리아에 가서 아이스크림을 사먹는데.c
#include <stdio.h>
#include <stdbool.h>

int main()
{
    int N, M;
    scanf("%d %d", &N, &M);
    bool bad[201][201] = {false};

    for (int i = 0; i < M; i++)
    {
        int a, b;
        scanf("%d %d", &a, &b);
        bad[a][b] = true;
        bad[b][a] = true;
    }

    int result = 0;
    for (int i = 1; i <= N; i++)
    {
        for (int j = i + 1; j <= N; j++)
        {
            if (bad[i][j])
                continue;
            for (int k = j + 1; k <= N; k++)
            {
                if (!bad[i][k] && !bad[j][k])
                {
                    result += 1;
                }
            }
        }
    }

    printf("%d\n", result);
    return 0;
}

File: 백준/Silver/2422. 한윤정이 이탈리아에 가서 아이스크림을 사먹는데/한윤정이 이탈리아에 가서 아이스크림을 사먹는데.py
import sys
input = sys.stdin.readline
N, M = map(int, input().split())
bad = set()
for _ in range(M):
    a, b = map(int, input().split())
    bad.add((a, b))
    bad.add((b, a))

result = 0
for i in range(1, N+1):
    for j in range(i+1, N+1):
        if (i, j) in bad:
            continue
        for k in range(j+1, N+1):
            if (i, k) in bad or (j, k) in bad:
                continue
            result += 1
print(result)
